src/ZendCodingStandard/Sniffs/Arrays/FormatSniff.php

```diff
@@ -0,0 +1,233 @@
+            return;
+        }
+
+        $this->singleLineArray($phpcsFile, $stackPtr);
+    }
+
+    /**
+     * @param File $phpcsFile
+     * @param int $stackPtr
+     * @return void
+     */
+    private function multiLineArray(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+        $arrayToken = $tokens[$stackPtr];
+
+        $bracketOpener = $arrayToken['bracket_opener'];
+        $bracketCloser = $arrayToken['bracket_closer'];
+
+        $firstContent = $phpcsFile->findNext(T_WHITESPACE, $bracketOpener + 1, null, true);
+        if ($tokens[$firstContent]['code'] === T_CLOSE_SHORT_ARRAY) {
+            $error = 'Empty array must be in one line.';
+            $fix = $phpcsFile->addFixableError($error, $stackPtr, 'EmptyArrayInOneLine');
+
+            if ($fix) {
+                $phpcsFile->fixer->replaceToken($bracketOpener + 1, '');
+            }
+
+            return;
+        }
+
+        $lastContent = $phpcsFile->findPrevious(T_WHITESPACE, $bracketCloser - 1, null, true);
+        if ($tokens[$bracketCloser]['line'] > $tokens[$lastContent]['line'] + 1) {
+            $error = 'Blank line found at the end of array';
+            $fix = $phpcsFile->addFixableError($error, $bracketCloser - 1, 'BlankLineAtTheEnd');
+
+            if ($fix) {
+                $phpcsFile->fixer->beginChangeset();
+                $i = $lastContent + 1;
+                while ($tokens[$i]['line'] !== $tokens[$bracketCloser]['line']) {
+                    $phpcsFile->fixer->replaceToken($i, '');
+                    ++$i;
+                }
+                $phpcsFile->fixer->addNewlineBefore($bracketCloser);
+                $phpcsFile->fixer->endChangeset();
+            }
+        }
+
+        $first = $phpcsFile->findFirstOnLine([], $bracketOpener, true);
+        $indent = $tokens[$first]['code'] === T_WHITESPACE
+            ? strlen($tokens[$first]['content'])
+            : 0;
+
+        $previousLine = $tokens[$bracketOpener]['line'];
+        $next = $bracketOpener;
+        while ($next = $phpcsFile->findNext(T_WHITESPACE, $next + 1, $bracketCloser, true)) {
+            if ($previousLine === $tokens[$next]['line']) {
+                if ($tokens[$next]['code'] !== T_COMMENT) {
+                    $error = 'There must be one array element per line.';
+                    $fix = $phpcsFile->addFixableError($error, $next, 'OneElementPerLine');
+
+                    if ($fix) {
+                        $phpcsFile->fixer->beginChangeset();
+                        if ($tokens[$next - 1]['code'] === T_WHITESPACE) {
+                            $phpcsFile->fixer->replaceToken($next - 1, '');
+                        }
+                        $phpcsFile->fixer->addNewlineBefore($next);
+                        $phpcsFile->fixer->endChangeset();
+                    }
+                }
+            } else {
+                if ($previousLine < $tokens[$next]['line'] - 1
```

Comment:
I would prefer to refactor it to extracting some parts to separate private methods instead of creating labels and goto instructions inside one method. What do you think? Do you want to have it before release or we can improve it later on? I will write new sniff with smaller methods, then it should be more readable. I try to add as much tests as I can to cover all cases. I think current code coverage is quite impressive 😄 

Even more, I would add sniff which will return error when `goto` instruction is used 😛